reactor-core/src/main/java/reactor/core/publisher/FluxConcatArray.java

```diff
@@ -503,7 +501,9 @@ public Object scanUnsafe(Attr key) {
 	static <T> void deactivateAndProduce(long produced, AtomicLongFieldUpdater<T> updater, T instance) {
 		for (;;) {
 			final long actualRequested = updater.get(instance);
-			final long deactivatedRequested = (actualRequested - produced) | Long.MIN_VALUE;
+			final long deactivatedRequested = actualRequested == Long.MAX_VALUE
+					? -1
```

Comment:
I'd prefer this to be more explicit, saves the (arguably small) mental gymnastic
```suggestion
					? Long.MAX_VALUE | Long.MIN_VALUE
```